src/App.jsx

```diff
@@ -1,61 +1,60 @@
+/* eslint-disable react-hooks/exhaustive-deps */
 import { useEffect, useState } from 'react';
-import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
-
 import { AddItem, Home, Layout, List } from './views';
-
+import { Routes, Route, useNavigate } from 'react-router-dom';
 import { getItemData, streamListItems } from './api';
 import { useStateWithStorage } from './utils';
+import { generateToken } from '@the-collab-lab/shopping-list-utils';
 
 export function App() {
+	const navigate = useNavigate();
 	const [data, setData] = useState([]);
-	/**
-	 * Here, we're using a custom hook to create `listToken` and a function
-	 * that can be used to update `listToken` later.
-	 *
-	 * `listToken` is `my test list` by default so you can see the list
-	 * of items that was prepopulated for this project.
-	 * You'll later set it to `null` by default (since new users do not
-	 * have tokens), and use `setListToken` when you allow a user
-	 * to create and join a new list.
-	 */
-	const [listToken, setListToken] = useStateWithStorage(
-		'my test list',
-		'tcl-shopping-list-token',
-	);
+
+	const [listToken, setListToken] = useStateWithStorage(null);
+
+	function handleClick() {
+		if (!listToken) {
+			const token = generateToken();
+			localStorage.setItem('tcl-shopping-list-token', token);
+			setListToken(token);
+			navigate('/list');
+		} else {
+			navigate('/list');
+		}
+	}
```

Comment:
This `handleClick` function should only be called when creating a new list so I don't think it's necessary to have an else that redirects to list.

Reply:
Fixed this!